### PR TITLE
Pseudolocalization

### PR DESCRIPTION
**Pull request by jeeyi**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /docker/.data
 /cli/target/
 /webapp/target/
+/webapp/package-lock.json
```
Comment (jeeyi) [inline .gitignore]: @aurambaj We don't need this file to be checked into git, correct?

Comment (aurambaj) [reply .gitignore]: yes it is needed it git, don't exclude it please.

File: webapp/src/main/java/com/box/l10n/mojito/service/assetintegritychecker/integritychecker/MessageFormatIntegrityChecker.java [hunk 2]
```diff
@@ -23,13 +20,15 @@
...
  */
-public class MessageFormatIntegrityChecker implements TextUnitIntegrityChecker {
+public class MessageFormatIntegrityChecker extends AbstractTextUnitIntegrityChecker {
 
     /**
      * logger
      */
     static Logger logger = LoggerFactory.getLogger(MessageFormatIntegrityChecker.class);
 
+    private static final String REPLACEMENT_CHARS = "!!";
```
Comment (aurambaj) [inline webapp/src/main/java/com/box/l10n/mojito/service/assetintegritychecker/integritychecker/MessageFormatIntegrityChecker.java]: might want to use some special unicode character to make it safer, like `\x13`

File: .gitignore (hunk 1 above)
Comment (jeeyi) [reply .gitignore]: Okay. I have reverted deletion of package-lock.json changes now.

File: webapp/src/main/java/com/box/l10n/mojito/service/assetintegritychecker/integritychecker/MessageFormatIntegrityChecker.java (hunk 2 above)
Comment (jeeyi) [reply webapp/src/main/java/com/box/l10n/mojito/service/assetintegritychecker/integritychecker/MessageFormatIntegrityChecker.java]: Good point!

**Comment by jeeyi** [comment]
@aurambaj I can't remember why I didn't use okapi code detection. Let me try with okapi. If it works, then it may be cleaner solution as you said.

**Comment by jeeyi** [comment]
@aurambaj Named arguments like `{username}` are not detected as inline code. Only indexed arguments like `{1}` are detected as expected. In order to make it work, we need to customize rules in codeFinder. This isn't simple for plural message format.

**Pull request merged**